### ISSUE TITLE
Release plan

### ISSUE THREAD

**Issue by felixfontein** [ansible-collections/community.routeros#1]
Small collections like this one don't need a complex plan like the one for community.general and community.network. So how about the following?

**Release minor and patch releases whenever we want (like after adding new features or fixing bugs). Since this collection is small, there's no need to fix things in advance. Just add features, and after a feature either wait a bit longer for more features/bugs, or make a release.**

I suggest releasing form `main` branch, as described here: https://github.com/ansible/community/wiki/ReleasingCollections#releasing-without-release-branches-for-smaller-collections

Once we release a 2.0.0 (with some breaking change relative to 1.x.y), we can have a `stable-1` branch so we can backport bugfixes (or even features) if needed, and release more 1.x.y versions.

(This is essentially what other collections I work on are doing, like community.crypto and community.sops.)

About the next release(s): I would suggest we quickly release a 1.0.0 version, so we can get it included in Ansible 2.10. We should do some testing with the current 0.1.0 release, maybe add bugfixes (or even features), if necessary release a 0.2.0 first, but not wait too long until 1.0.0.

What do you think?

CC @renatoalmeidaoliveira @NikolayDachev @adeptvin1 @heuels

(BTW, I put @heuels and @NikolayDachev as the authors of this collection since they created the plugins and modules in it. I hope that's ok for everyone!)

**Comment by gundalow** [comment]
@felixfontein thank you for doing this.

@heuels and @NikolayDachev please check your emails for an invite to join this repo, and get commit.

**Comment by NikolayDachev** [comment]
Hey, 

Thanks ! 

Will check  the release notes (I guess @felixfontein suggestion is ok  ) also will try to check / test the repo and api module this days

**Comment by NikolayDachev** [comment]
I have a small collection with different roles which are still in progress (total lack of free time). I switch all roles tasks to use community.routeros.api and they look ok.

_just a thought:_
RouterOS api is new thing however commands and facts probably are already used in lot of users roles. This change to community.routeros.* probably will make some users to be little anoyed about it. For me was easy to do that in a single cmd for all roles/tasks with help of grep,awk and sed  etc. but ..
My point is, is there any user guidelines for similar changes from the past?

**Comment by NikolayDachev** [comment]
About release plan:

`About the next release(s): I would suggest we quickly release a 1.0.0 version `

When we can / we should do that (as a date/time frame)?

**Comment by felixfontein** [comment]
> I have a small collection with different roles which are still in progress (total lack of free time). I switch all roles tasks to use community.routeros.api and they look ok.

:+1:

> _just a thought:_
> RouterOS api is new thing however commands and facts probably are already used in lot of users roles. This change to community.routeros.* probably will make some users to be little anoyed about it. For me was easy to do that in a single cmd for all roles/tasks with help of grep,awk and sed etc. but ..
> My point is, is there any user guidelines for similar changes from the past?

There aren't any guidelines yet, since this situation is pretty new :) What's needed to do depends on the version of Ansible that is used.

For Ansible 2.9, which comes with included routeros_* modules (routeros_command and routeros_facts), you have to explicitly use FQCNs to use community.network.routeros_*. These will stop working when users install community.network 2.0.0 (since the modules will have been removed by then), and they manually have to install community.routeros and update their FQCNs (via search+repleace).

For ansible-base 2.10 users which explicitly installed community.network, they either use the old shortname routeros_* or the FQCN community.network.routeros_*. Once they upgrade to community.network 2.0.0, they will get errors **if they do not install community.routeros**. Once they install community.routeros, thanks to the redirects in community.network their roles and playbooks will just continue to work. If they ever get rid of community.network and just want to use community.routeros though, then they have to update their FQCNs. (Possibilty upgrading to ansible-base 2.11 will solve this, as its ansible_builtin_runtime.yml might get updated once last time, if they are still using the shortname routeros_* from Ansible 2.9 and earlier.)

For Ansible 2.10 users (Ansible 2.10 = ansible-base 2.10 + a set of collections) who upgrade to Ansible 2.11, they have to do nothing. It will just continue to work and use the modules from community.routeros. Ansible 2.11 (ansible-base 2.10 + a set of collections) will include community.network and community.routeros, and thanks to the redirects in community.network both the short names routeros_* and the community.network FQCNs will work. (Ansible 2.12 will probably include ansible-base 2.11, then the shortnames routeros_* will redirect directly to community.routeros.*.)

> About release plan:
> 
> `About the next release(s): I would suggest we quickly release a 1.0.0 version `
> 
> When we can / we should do that (as a date/time frame)?

We can do that today if we want :) I'd suggest to wait a bit, maybe two weeks, just to see if something unexpected shows up.

I'd like to make a 0.1.1 release soon, once ansible-collections/community.network#138 has been merged and 'foreported' to this repo (I hope in the next days). If anyone finds other things to fix/improve, please create PRs :)

**Comment by NikolayDachev** [comment]
Thanks for the info!

`We can do that today if we want :) I'd suggest to wait a bit, maybe two weeks, just to see if something unexpected shows up.`

I fully agree with that

**Comment by felixfontein** [comment]
Since ansible-collections/community.network#138 isn't making any progress, I'll release 0.1.1 now.

**Comment by felixfontein** [comment]
[community.routeros 0.1.1](https://galaxy.ansible.com/community/routeros) has been released.

**Comment by felixfontein** [comment]
What do you think about releasing 1.0.0 of this collection in ~a week? I have two open PRs which should be merged / closed until then.

In case you found anything during testing so far, please report it.

**Comment by NikolayDachev** [comment]
I have a 0 time to check them this days will see if I will be able to find a time ( work,baby and second 4 years old kid .. :) )to help with them .. I think we shuld be ok for 1.0...

**Comment by heuels** [comment]
I hope to find time to test some more on the weekend, but as of now everything looks OK 🙂

**Comment by NikolayDachev** [comment]
look like the 1st PRs @heuels already work on it .. if need a help to speed up the fix, let me know, the second one look like is going to nowhere so I guess we should be good to go for the next week ?

**Comment by felixfontein** [comment]
@NikolayDachev which PRs do you mean?

**Comment by NikolayDachev** [comment]
sorry for confusion **issues** not PRs

**Comment by felixfontein** [comment]
Haha, ok, that makes a lot more sense :) Yes I agree. Once we released 1.0.0 we can also release 1.0.1 whenever we fixed a bug, there's no need to wait for two months for the next release as in community.network ;)

**Comment by NikolayDachev** [comment]
I think (if we all agree) next Tuesday is a great date for 1.0.0

**Comment by felixfontein** [comment]
Next Tuesday is 1.0.0 day then (if nothing comes up over the weekend)! :)

**Comment by NikolayDachev** [comment]
> Next Tuesday is 1.0.0 day then (if nothing comes up over the weekend)! :)

Yep Friday is always a bad day for releases or changes ... Monday, I think a majority of the people just not like it for obvious reasons :)

**Comment by felixfontein** [comment]
No complaints, and the nightly CI didn't fail, so doing the 1.0.0 relesae now :tada:

**Comment by felixfontein** [comment]
1.0.0 has been released: https://galaxy.ansible.com/community/routeros

**Comment by felixfontein** [comment]
PR for inclusion in Ansible 2.10: ansible-community/ansible-build-data#35

**Comment by gundalow** [comment]
@felixfontein @NikolayDachev @heuels great work everybody on reaching this major milestone!

@NikolayDachev @heuels If you haven't already, please subscribe to the Ansible Bullhorn email list https://us19.campaign-archive.com/home/?u=56d874e027110e35dea0e03c1&id=d6635f5420 this newsletter contains important information for contributors.

**Comment by felixfontein** [comment]
1.0.1 with #15 has just been released.

**Comment by felixfontein** [comment]
Looks like something went wrong during releasing 1.1.0 (https://dashboard.zuul.ansible.com/t/ansible/stream/b6021035e749489a8e445e1c471986c1?logfile=console.log). I'll ping people on IRC about that...

**Comment by felixfontein** [comment]
Ok, it did complete. Just took longer (and had a lot of errors in between - unrelated to us though). The uploaded artefact to galaxy looks good, so it should be fine :)

**Comment by felixfontein** [comment]
@heuels I think we should do 1.2.0 soon (as in: end of this week), with #36 and #37. We also have a new feature waiting since end of March (#23)... The next Ansible release will be 4.2.0 beginning of next week, so the new version would get included in there. Does that sound good?

**Comment by heuels** [comment]
@felixfontein Yes, sounds good — let's release 1.2.0 👍🏻

**Comment by felixfontein** [comment]
1.2.0 has been released! :tada:

**Comment by NikolayDachev** [comment]
Should we do 1.2.1 after #39 is merge ? Overall is important fix

**Comment by felixfontein** [comment]
I think we should, but we should make sure that the fix and changelog fragment are good :)

**Comment by felixfontein** [comment]
I've created a [stable-1 branch](https://github.com/ansible-collections/community.routeros/tree/stable-1) and [bumped the main branch to 2.0.0](https://github.com/ansible-collections/community.routeros/commit/b45fd9415f8a897969013fba4ae57e71e92fc3fa).

**Comment by felixfontein** [comment]
Should we create a 2.0.0-a1 prerelease, do some testing without a prerelease, or proceed directly to 2.0.0?

**Comment by NikolayDachev** [comment]
I'm not sure what you mean with "do some testing without a prerelease" ?

About releasing personally I'm more conservative, aka I prefer to no go direct with main release , from other side we should avoid bunch of pre-releases ....

**Comment by felixfontein** [comment]
> I'm not sure what you mean with "do some testing without a prerelease" ?

You can also test the collection when installing it from git, or when checking it out in a path where ansible-core will find it. I'm doing it for example and am using its `main` branch. I don't really use this collection most of the time though, so I'm not actively testing it :) But someone else could test it this way.

> About releasing personally I'm more conservative, aka I prefer to no go direct with main release , from other side we should avoid bunch of pre-releases ....

I would also wait a bit more with the final 2.0.0 release. I'm not sure how many people would acually try out a prerelease instead of installing the collection's current `main` branch from git...

**Comment by NikolayDachev** [comment]
> I would also wait a bit more with the final 2.0.0 release. I'm not sure how many people would acually try out a prerelease instead of installing the collection's current `main` branch from git...

Unfortunately the reality is .. 80% or more of the people always skip reading .. "they pull" after that they open PR's .. :) How we can solve this is a not easy task. 

Let's make a prerelease, I don't think we will have a lot, so we can go with the "safe side"

**Comment by felixfontein** [comment]
Sounds good. I'll create a pre-release the next days then.

**Comment by felixfontein** [comment]
I've just released 2.0.0-a1.

**Comment by felixfontein** [comment]
The Bullhorn 32 (https://mailchi.mp/redhat/the-bullhorn-32) has been sent a few days ago, which contains an announcement of 2.0.0-a1 and which asks for feedback. I'd suggest to wait a couple of weeks more, and if nobody reacts until then, make the 2.0.0 release.

**Comment by NikolayDachev** [comment]
I don't see any reason to not do that so  let's go with  2.0.0-a2

**Comment by felixfontein** [comment]
Should we plan to release 2.0.0 by the end of this month, if nothing serious shows up until then? We should have 2.0.0 released by November 8th so that it gets included in Ansible 5, but of course we can also release it before that :)

**Comment by felixfontein** [comment]
I guess that since nobody replied, it's ok if I do the release today.

**Comment by felixfontein** [comment]
2.0.0 has been released! :tada:

**Comment by NikolayDachev** [comment]
> 2.0.0 has been released! 🎉

Great news ! 
Congrats all for 2.0.0

**Comment by NikolayDachev** [comment]
Wish you all very merry Christmas and happy new 2022 :)

**Comment by felixfontein** [comment]
@NikolayDachev also Merry Christmas a Happy New Year to you, and to everyone else! Enjoy the new year, let's hope it has better surprises than the previous two ones :)

**Comment by felixfontein** [comment]
2.1.0 has been released with a ton of new features!

**Comment by felixfontein** [comment]
2.2.0 has been released with two new modules and improved licensing marking.

**Comment by felixfontein** [comment]
2.2.1 has been released with some bugfixes for the new modules.

**Comment by felixfontein** [comment]
2.3.0 has been released with some bugfixes and new features.

**Comment by NikolayDachev** [comment]
@felixfontein Thank you !!!

**Comment by bigdestroyer** [comment]
Curious, can one run with one play the playbook to multiple devices at once? I would like to make an inventory file with all the mikrotik devices inside and run the playbook against it. Would help for bulk provisioning/applying changes. Looking at the examples it seems you can only do 1 router per play before having to change the playbook.

**Comment by felixfontein** [comment]
@bigdestroyer please do not post questions in issues that are used for something completely different. I've locked this issue so this won't happen again.

To answer your question (please create a new issue if you want to discuss this in more detail): of course you can do that. The examples mainly try to show how something works in a simple case, so they stick to a simple(r) operation for one router. But you can easily run them against multiple routers, switches, or other MikroTik devices at the same time.

**Comment by felixfontein** [comment]
2.3.1 has been released with improved documentation.

**Comment by felixfontein** [comment]
2.4.0 is out with quite a few improvements for the API modules.

**Comment by felixfontein** [comment]
2.5.0 is out with new features and a bugfix for the API modules.

**Comment by NikolayDachev** [comment]
@felixfontein Thank you !!!

**Comment by felixfontein** [comment]
@NikolayDachev thanks! @therfert also deserves a lot of credit for quite a few of the new feature/bugfix PRs!

**Comment by felixfontein** [comment]
2.6.0 has been released with new features and bugfixes.

**Comment by felixfontein** [comment]
1.2.1 and 2.7.0 are out. I noticed that we backported some bugfixes to the stable-1 branch some longer time ago, but never released them; 1.2.1 contains them. 2.7.0 has a fix and new features for the new api modules.

**Comment by NikolayDachev** [comment]
Thank you !

**Comment by felixfontein** [comment]
2.8.0 is out with new features and bugfixes! Thanks to everyone who contributed!

**Comment by felixfontein** [comment]
2.8.1 is out with a bugfix.

**Comment by felixfontein** [comment]
2.8.2 is out with another bugfix.

**Comment by felixfontein** [comment]
2.8.3 has been released with updated documentation that now uses semantic markup.

**Comment by felixfontein** [comment]
2.9.0 has been released with support for new API paths and fixed support for existing paths.

**Comment by felixfontein** [comment]
2.10.0 is out with a bunch of new features w.r.t. API paths!

**Comment by felixfontein** [comment]
2.11.0 has been released with new features and bugfixes.

**Comment by felixfontein** [comment]
If anyone still uses the 1.x.y versions, please take a look at #252. I'm proposing to declare the stable-1 branch as End Of Life.

**Comment by felixfontein** [comment]
2.13.0 has been released with new features and bugfixes.

**Comment by felixfontein** [comment]
1.2.2 has been released, making the stable-1 branch effectively End of Life. There will be no further 1.x.y release. (Ref: #252)

**Comment by felixfontein** [comment]
2.14.0 is out with new features.

**Comment by felixfontein** [comment]
2.15.0 is out with more new features!

**Comment by felixfontein** [comment]
2.16.0 is out with more new features!

**Comment by felixfontein** [comment]
2.17.0 is out with new features!

**Comment by felixfontein** [comment]
2.18.0 is out with a bugfix, several new features (including a `restrict` feature for api_info and api_modify), and a deprecation of support for ansible-core < 2.15 for the next major release.

**Comment by felixfontein** [comment]
2.20.0 is out. Tomorrow I'll merge the 3.0.0 PR (#318) and then someone next week there'll be 3.0.0 with some breaking changes.

**Comment by felixfontein** [comment]
3.0.0 is out, which removes support of EOL Ansible/ansible-base/ansible-core versions and fixes check mode of the community.routeros.command module. The 2.x.y release stream will only receive bugfix releases (2.20.x) from now on.

**Comment by felixfontein** [comment]
3.1.0 is out with bugfixes and features.

**Comment by felixfontein** [comment]
3.2.0 is out with a new feature and updated documentation.

**Comment by felixfontein** [comment]
3.3.0 is out with new features.

**Comment by felixfontein** [comment]
3.4.0 is out with a new feature and a bugfix.

**Comment by felixfontein** [comment]
3.5.0 is out with a new feature.

**Comment by felixfontein** [comment]
2.20.1 has been released to mark community.routeros 2.x.y as End of Life.

**Comment by felixfontein** [comment]
3.6.0 is out with new features for the API modules.

**Comment by felixfontein** [comment]
3.8.0 is out with new features for the API modules.

**Comment by felixfontein** [comment]
3.9.0 is out with a bugfix and some new features

**Comment by felixfontein** [comment]
3.10.0 is out with a bugfix and new features.

**Comment by felixfontein** [comment]
3.11.0 is out with a bugfix and a new feature.

**Comment by felixfontein** [comment]
3.15.0 is out with new features for the API modules, and one deprecation.

**Comment by felixfontein** [comment]
3.17.0-a1 is out with a huge amount of changes to the API modules (`api_modify` and `api_info`), thanks to @FelixRicke (with PR #433)! Please carefully test this release against existing playbooks/roles and report any problems you encounter (that you think need to be fixed), so we can confidently create a 3.17.0 release (or more pre-releases if necessary).

Pinging @0xUnicorn @20250414 @chaeynz @derdeagle @DonGiovanni83 @ewysong @hansmi @ivoruetsche @janbue @jraylan @limjh16 @Marcusz97 @Paulomart @petrows @prinzdezibel @pypb @radokristof @rondex @SdVSysAdmin @sgd-gh @therfert @tim427 @Tr4sK @tumd @vfreex @YannSK since you contributed (by creating PRs or issues) to this collection in 2025 and/or 2026, and hopefully still use it. Remember to make sure to run your playbooks with `--check` when trying this version.

**Comment by felixfontein** [comment]
[3.17.0-a2 is out with a few fixes](https://github.com/ansible-collections/community.routeros/blob/main/CHANGELOG.md#v3-17-0-a2). Please continue to test this :)

**Comment by felixfontein** [comment]
3.19.0 is out with more improvements and some deprecations.

**Comment by felixfontein** [comment]
3.20.0 is out with more API module features.

**Comment by felixfontein** [comment]
3.21.0 is out with more API module features.

**Comment by felixfontein** [comment]
3.22.0 is out with more API module features.
